vcs/gitcmd/repo.go

```diff
@@ -161,105 +161,123 @@ func (r *Repository) ResolveTag(name string) (vcs.CommitID, error) {
 	return commitID, nil
 }
 
-// branchCounts returns the behind/ahead commit counts information for branch, against base branch.
-func (r *Repository) branchCounts(branch, base string) (behind, ahead uint, err error) {
-	if err := checkSpecArgSafety(branch); err != nil {
-		return 0, 0, err
-	}
-	if err := checkSpecArgSafety(base); err != nil {
-		return 0, 0, err
-	}
+// stringFilter helps manage a list of string filters.
+type stringFilter struct{ only map[string]struct{} }
```

Comment:
Great. :)

I like `branchFilter` more since it limits the scope to one specific task. There's no need for it to be a `stringFilter` since it's not used anywhere else for filtering strings, so might as well take advantage of that and increase specialization. More general code is harder to work with, so it's best to avoid unless you benefit from it being general - which isn't the case here.